### PR TITLE
feat: resolve minimap tile urls in map component

### PR DESCRIPTION
**Pull request by fallenoak**
This PR removes the redirect mechanism to resolve minimap tile URLs in favor of a tile index API that serves up the mapping of tile path to content path for each tile directory in the `trs` file.

The immediate benefit is fewer HTTP requests when rendering minimaps in `spelunker-web`. The next phase of this work will involving switching to `@wowserhq/format`'s `Blp` class to render tile `.blp` files directly in `spelunker-web`.

### CODE REVIEW

File: packages/spelunker-api/src/lib/minimap/tiles.mjs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+for (const line of source.split(/\n|\r\n/)) {
+  if (line.startsWith('dir:') || line.trim().length === 0) {
+    continue;
+  }
+
+  const [tilePath, contentPath] = line.split(/\t+/);
+  const tilePathParts = tilePath.split(/[/\\]/);
+  const tileDirectory = tilePathParts.slice(0, -1).map((t) => t.toLowerCase()).join('/');
+  const tileName = tilePathParts.at(-1).split('.')[0].toLowerCase();
```
Comment (timkurvers) [inline packages/spelunker-api/src/lib/minimap/tiles.mjs:24]: Not entirely sure but have a feeling we may be able to piggyback on `path.posix` or `path.win32` here to do the heavy lifting of converting between one and the other.

File: packages/spelunker-api/src/lib/pipeline/index.mjs [hunk 2]
```diff
@@ -25,19 +24,6 @@ pipeline.param('file', (req, res, next, path) => {
...
-  const { mapName, tx, ty } = req.params;
-  const index = `${mapName}\\map${tx}_${ty}`;
-  const tile = minimapTiles[index];
-  if (tile) {
-    res.redirect(`${req.baseUrl}/files/textures/minimap/${tile}.png`);
-  } else {
-    res.sendStatus(404);
-  }
-});
-
```
Comment (timkurvers) [inline packages/spelunker-api/src/lib/pipeline/index.mjs:40]: 👍

File: packages/spelunker-web/src/components/core/GameMap/index.jsx [hunk 3]
```diff
@@ -18,24 +19,58 @@ const crs = Object.assign({}, CRS.Simple, {
...
+
 class MinimapTileLayer extends TileLayer {
   getTileUrl({ x, y }) {
     const tx = 32 + x;
     const ty = 32 + y;
-    const mapName = this._url;
-    return `${process.env.PIPELINE_URI}/minimap/${mapName}/${tx}/${ty}.blp.png`;
+
+    // TODO use real url
+    const unknownTileUrl = `${process.env.PIPELINE_URI}/files/textures/minimap/unknown_${tx}_${ty}`;
```
Comment (timkurvers) [inline packages/spelunker-web/src/components/core/GameMap/index.jsx:41]: Given the TODO here, is this a 404 at present? Could this be a generic URL that does not need `tx` and `ty`?

**Review by timkurvers — APPROVED** [review]
🥳

File: packages/spelunker-web/src/components/core/GameMap/index.jsx (hunk 3 above)
Comment (fallenoak) [reply packages/spelunker-web/src/components/core/GameMap/index.jsx:41]: Yeah, this is an intentional 404. The `tx` and `ty` is present just to make it easier to trace while we're monkeying a bunch with `GameMap`.

Do you know if Leaflet accepts `null` responses from `getTileUrl`? I haven't looked too closely yet.

File: packages/spelunker-api/src/lib/minimap/tiles.mjs (hunk 1 above)
Comment (fallenoak) [reply packages/spelunker-api/src/lib/minimap/tiles.mjs:24]: I'm trying to recall if `MPQ`s are path delimiter agnostic or not. If not, I'd agree that we could probably use `path.win32.parse` to extract the directory and file name. Regardless, I suspect the `trs` file is always written using `\`, so `path.win32.parse` likely always works.

**Pull request merged**